src/app/ipv/middleware.js

```diff
@@ -156,19 +159,13 @@ async function handleBackendResponse(req, res, backendResponse) {
   throw new Error(message.description);
 }
 
-function tryValidateCriResponse(criResponse) {
-  if (!criResponse?.redirectUrl) {
-    throw new Error("CRI response RedirectUrl is missing");
-  }
-
-  return true;
-}
-
-function tryValidateClientResponse(client) {
-  const { redirectUrl } = client;
+function validateRedirectUrlInResponse(response, isClient = false) {
+  const { redirectUrl } = response;
 
   if (!redirectUrl) {
-    throw new Error("Client Response redirect url is missing");
+    throw new Error(
+      `${isClient ? "Client" : "CRI"} Response RedirectUrl is missing`,
```

Comment:
Does it want to be specific about where the response comes from? I imagine it would be obvious from context but maybe I'm wrong?

Reply:
We receive response from core-back, CRI is from CRI provider and cleint not 100% sure

Reply:
The refactoring might not be worth it if we want to keep the log message specific to the source then... I imagined that "Response RedirectUrl is missing" would be fine since given context you'd know where it'd come from?

Reply:
Client is orchestration 
It should be clear from a sequence of logs in a journey yes